src/Meshes/mesh.ts

```diff
@@ -2021,7 +2021,7 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
             if (batch.renderSelf[subMesh._id]) {
                 // Draw
                 if (onBeforeDraw) {
-                    onBeforeDraw(false, renderingMesh._effectiveMesh.getWorldMatrix(), effectiveMaterial, renderingMesh._effectiveMesh);
+                    onBeforeDraw(false, renderingMesh.getWorldMatrix(), effectiveMaterial);

```

Comment:
I'm not confident this code is right. Can someone look over this and see if it makes sense? Also, what's a good way to test this code?

Reply:
This code path is the normal code path when rendering a mesh without instances, so it can be tested with the default playground. Now, I'm not sure if it is `renderingMesh` that should be used to get the world matrix...

Reply:
renderingMesh._effectiveMesh will return renderingMesh so I guess we are fine